helper.py

```diff
@@ -1,41 +1,186 @@
-                scale_y = out_height / height
-                if scale_x > scale_y:
-                    img = img.resize((int(width*scale_y), int(height*scale_y)), Image.ANTIALIAS)
-                else:
-                    img = img.resize((int(width*scale_x), int(height*scale_x)), Image.ANTIALIAS)
-
-                mod_time = stat_obj.st_mtime
-                img.save(output_file,"JPEG")
-                with open(stat_file, "w") as myfile:
-                    myfile.write(str(mod_time))
-            except IOError:
-                pass
+log_static_path = "/media/RobotUSB/"
+log_input_file = log_static_path + "logs.txt"
+
+picture_watcher.watch(
+    alias="image", path=img_input_file, flags=aionotify.Flags.MODIFY
+)  # sets up watcher
+log_watcher.watch(alias="logs", path=log_input_file, flags=aionotify.Flags.MODIFY)
+
+
+def shrink_image(img):
+    width_i, height_i = img.size
+    width = float(width_i)
+    height = float(height_i)
+    scale_x = out_width / width
+    scale_y = out_height / height
+    if scale_x > scale_y:
+        img = img.resize((int(width * scale_y), int(height * scale_y)), Image.LANCZOS)
+    else:
+        img = img.resize((int(width * scale_x), int(height * scale_x)), Image.LANCZOS)
+    return img
+
+
+def im_2_b64(image):
+    buff = BytesIO()
+    image.save(buff, format="JPEG")
+    img_str = base64.b64encode(buff.getvalue())
+    return img_str
+
+
+async def wait_for_picture_change():
+    loop = asyncio.get_event_loop()
+
+    bypass = False  # so first image is not ignored.
+    while not os.path.exists(img_input_file):
+        await asyncio.sleep(0.5)  # twiddle thumbs :)
+        if not bypass:
+            bypass = True
+    await picture_watcher.setup(loop)
+    print("Image change watcher is running.")
+
+    while True:  # for all events
+        if not bypass:
+            event = await picture_watcher.get_event()  # blocks until file changed
+        else:
+            bypass = False  # reset bypass
+
+        for c in range(10):
+            await asyncio.sleep(0.01)  # give it time to write the file.
+            try:  # this runs until the bot has finished writing the image
+                img = Image.open(img_input_file)
+                img.load()
+                print("Opened successfully")
+                break
+            except:
+                print("Error opening file: attempt \#" + str(c))
+
+        if c >= 9:
+            continue  # error with this file, go back and wait for next change.
+
+        img = shrink_image(img)
+        # converts image to a base64 which can be sent as a long string to the browser.
+        img_b64 = im_2_b64(img).decode()
+        websockets.broadcast(
+            CONNECTIONS, "[CAMERA]" + img_b64
+        )  # sends image to all connected browsers.
+        print("Image broadcast.")
+
+    # politely stops watching file system.
+    picture_watcher.close()
+    loop.stop()
+    loop.close()
+
+
+async def wait_for_log_change():
```

Comment:
There is a lot of redundancy between this and `wait_for_picture_change` possibly can make the code easier to read and update by combining these?

Reply:
I could do, it just makes it easier to change aspects of how the log part works... As they have key differences with not needing to resize image etc so it was easier to separate out than combine